routing/turns_sound.cpp

```diff
@@ -44,14 +44,39 @@ string TurnsSound::GenerateTurnText(uint32_t distanceUnits, uint8_t exitNum, boo
   return m_getTtsText(notification);
 }
 
-void TurnsSound::GenerateTurnSound(TurnItem const & turn, double distanceToTurnMeters,
-                                   vector<string> & turnNotifications)
+void TurnsSound::GenerateTurnSound(vector<TurnItemDist> const & turns, vector<string> & turnNotifications)
 {
   turnNotifications.clear();
 
-  if (!m_enabled)
+  if (!m_enabled || turns.empty())
     return;
 
+  TurnItemDist const & firstTurn = turns.front();
+  string firstNotification = GenerateFirstTurnSound(firstTurn.m_turnItem, firstTurn.m_distMeters);
+  if (m_nextTurnNotificationProgress == PronouncedNotification::Nothing)
+    return;
+  if (firstNotification.empty())
+    return;
+  turnNotifications.emplace_back(move(firstNotification));
+
+  // Generating notifications like Then turn left if necessary.
+  if (turns.size() <= 1)
```

Comment:
```c++
if (turns.size() < 2)
  return;
```
I know, it's equivalent to what's already written, but it's looks more clear as you introduce "secondTurn" on the next line.